rearrange_cars_luiza.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jul 22 10:37:40 2017
+Rearrange the car assignment
+@author: Luiza
+"""
+
+import logging
+import unittest
+import numpy as np
+
+'''Given: There is a parking lot with N spaces and N-1 cars in it.
+The parking lot is described by an array of numbers, that is a permutation of numbers
+from 0 to N. Cars are with numbers from 1 to N-1, 0 means an empty parking space.
+
+E.G. Initial order = [1, 2, 0, 3]
+       Final order = [3, 1, 2, 0]
+
+Task: to rearrange the cars in a given way
+and print a series of moves from position x to y
+
+For our example, the series of moves should be e.g.:
+    move 1 to 2
+    move 3 to 0
+'''
+
+class RearrangeCars(object):
+
+    '''Class to rearrange the cars and print the moves
+       - Uses car_position_dict to optimize searching the position of a given car
+       - Stores additionally the position of empty spot in current car arrangement
+       self.current_car_order.
+       - Moves are either printed to the console or logged
+       - Stores and updates min_num_steps_left that shows min number of moves left
+        in order to avoid comparison current and final order lists that takes O(n)
+    '''
+
+    def __init__(self, init_car_order, final_car_order):
+
+        '''Initializes class instance with initial, final order and
+        stores the position of empty parking space in the current car order
+
+        Arguments:
+           init_car_order: a valid non-empty list of cars + 1 \
+           empty_place given in their parking place order
+           
+           final_car_order: a valid non-empty list of cars + 1 \
+           empty_place according to the desired order
+
+        Returns:
+            None
+        '''
+
+        self.initial_car_order = init_car_order
+        self.current_car_order = init_car_order
+        self.final_car_order = final_car_order
+        self.empty_spot_position = self.current_car_order.index(0)
+        self.print_console = True
+        self.car_position_dict = {car:pos for pos, car in enumerate(init_car_order) \
+                                  if car != 0} #search the index of car in O(1) via dict
+        self.min_num_steps_left = None
+
+    def print_one_step(self, from_idx):
+
+        '''Prints a move of the car from position #from_idx to empty spot
+        '''
+
+        move_str = 'move from {0} to {1}'.format(from_idx, self.empty_spot_position)
+        if self.print_console:
+            print move_str
+        else:
+            logging.info(move_str)
+
+    def create_log_file(self):
+
+        '''In case the number of moves is more than 100
+        writes the output to the logfile instead of the console
+        '''
+
+        logging.warning('Since the number of cars to arrange is {0} and more than 1000\

```

Comment:
Unfortunately I never worked with logging before so I cannot provide much advice for this function, just what is the \^M for? And what you mean by "and more than 1000"? It feels like the sentence in incomplete.